File: status-layout/StatusLibrary/src/main/java/com/status/layout/Status.java
package com.status.layout;

import android.support.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * @author y
 */
@StringDef({
        Status.NORMAL,
        Status.LOADING,
        Status.EMPTY,
        Status.SUCCESS,
        Status.ERROR
})
@Retention(RetentionPolicy.SOURCE)
public @interface Status {
    String NORMAL = "StatusLayout:Normal";
    String LOADING = "StatusLayout:Loading";
    String EMPTY = "StatusLayout:Empty";
    String SUCCESS = "StatusLayout:Success";
    String ERROR = "StatusLayout:Error";
}

File: status-layout/StatusLibrary/src/main/java/com/status/layout/Status.kt
package com.status.layout

import android.support.annotation.StringDef

/**
 * @author y
 */
object Status {
    const val NORMAL = "StatusLayout:Normal"
    const val LOADING = "StatusLayout:Loading"
    const val EMPTY = "StatusLayout:Empty"
    const val SUCCESS = "StatusLayout:Success"
    const val ERROR = "StatusLayout:Error"

    @StringDef(Status.NORMAL, Status.LOADING, Status.EMPTY, Status.SUCCESS, Status.ERROR)
    @Retention(AnnotationRetention.SOURCE)
    annotation class StatusAnnotation
}
